lib/modules/filteReddit.js

```diff
@@ -12,21 +14,609 @@ import {
+				remainder /= multiplier;
+				remainderQualifier = qualifier;
+			} while (qualifiers.length);
+		},
+	},
+	postAfter: {
+		name: 'Post after',
+		trueText: 'after',
+		falseText: 'before',
+		defaultTemplate(patt) {
+			return { type: 'postAfter', patt: patt || '' };
+		},
+		fields: [
+			'posted after date ',
+			{ type: 'text', id: 'patt', validator: v => { if (isNaN(Date.parse(v))) throw Error('Could\'t parse'); } },
+		],
+		evaluate(thing, data) {
+			const postTime = thing.getTimestamp();
+			if (!postTime) return false;
+
+			return postTime && postTime >= new Date(data.patt);
+		},
+		pattern: 'A string representing a RFC2822 or ISO 8601 date',
+		parse(input) {
+			const date = Date.parse(input);
+			return isNaN(date) ? null : this.defaultTemplate(input);
+		},
+		fromSelected(thing) { return thing.getTimestamp().toISOString(); },
+	},
+	postTitle: {
+		name: 'Post title',
+		trueText: 'title contains',
+		falseText: '¬ title contains',
+		defaultTemplate(patt) {
+			return { type: 'postTitle', patt: patt || '' };
+		},
+		fields: [
+			'post\'s title contains ', { type: 'text', id: 'patt', validator: RegExp },
+		],
+		evaluate(thing, data) {
+			const title = thing.getTitle();
+			// Do not anchor for this case
+			return new RegExp(data.patt, 'i').test(title);
+		},
+		parse(input) { return this.defaultTemplate(input); },
+	},
+	postType: {
+		name: 'Post type',
+		trueText: 'type',
+		falseText: '¬ type',
+		defaultTemplate(kind) {
+			return { type: 'postType', kind: kind || 'link' };
+		},
+		fields: [
+			'post is a ', {
+				// id: 'type' results in a collsion
+				type: 'select',
+				id: 'kind',
+				options: [
+					['link post', 'link'],
+					['self post', 'self'],
+					// 'comment',
+				],
+			}, '.',
+		],
+		evaluate(thing, data) {
+			switch (data.kind) {
+				case 'comment': return thing.isComment();
+				case 'link':    return thing.isLinkPost();
+				case 'self':    return thing.isSelfPost();
+				default: return false;
+			}
+		},
+		pattern: 'x — where x is any of the following: comment, link, self',
+		parse(input) { return this.defaultTemplate(input); },
+	},
+	isNSFW: {
+		name: 'NSFW post',
+		unique: true,
+		trueText: 'nsfw',
+		falseText: '¬ nsfw',
+		get alwaysShow() { return !module.options.NSFWfilter.value; },
+		defaultTemplate() {
+			return { type: 'isNSFW' };
+		},
+		fields: [
+			'post is marked NSFW',
+		],
+		evaluate(thing) {
+			return thing.isNSFW();
+		},
+		parse() { return this.defaultTemplate(); },
+	},
+	isVisited: {
+		/* Not available for Safari and Edge, as they lack isURLVisited */
+		name: 'visited',
+		unique: true,
+		trueText: 'visited',
+		falseText: '¬ visited',
+		defaultTemplate() {
+			return { type: 'isVisited' };
+		},
+		fields: [
+			'post link has not been visited',
+		],
+		disabled: process.env.BUILD_TARGET === 'safari' || process.env.BUILD_TARGET === 'edge',
+		get alwaysShow() { return !this.disabled; },
+		async evaluate(thing) {
```

Comment:
Either remove `async` (which is unnecessary currently) or `await` `isURLVisited` (don't care which)